azure-sink-connector/src/main/java/io/aiven/kafka/connect/azure/AzureBlobSinkConfig.java

```diff
@@ -0,0 +1,307 @@
+                }, ConfigDef.Importance.LOW, "Specifies the timestamp variable source. Default is wall-clock.",
+                GROUP_FILE, fileGroupCounter, ConfigDef.Width.SHORT, FILE_NAME_TIMESTAMP_SOURCE);
+    }
+
+    public AzureBlobSinkConfig(final Map<String, String> properties) {
+        super(configDef(), handleDeprecatedYyyyUppercase(properties));
+        validate();
+    }
+
+    static Map<String, String> handleDeprecatedYyyyUppercase(final Map<String, String> properties) {
+        if (properties.containsKey(FILE_NAME_TEMPLATE_CONFIG)) {
+            final var result = new HashMap<>(properties);
+
+            String template = properties.get(FILE_NAME_TEMPLATE_CONFIG);
+            final String originalTemplate = template;
+
+            final var unitYyyyPattern = Pattern.compile("\\{\\{\\s*timestamp\\s*:\\s*unit\\s*=\\s*YYYY\\s*}}");
+            template = unitYyyyPattern.matcher(template)
+                    .replaceAll(matchResult -> matchResult.group().replace("YYYY", "yyyy"));
+
+            if (!template.equals(originalTemplate)) {
+                LOG.warn(
+                        "{{timestamp:unit=YYYY}} is no longer supported, "
+                                + "please use {{timestamp:unit=yyyy}} instead. " + "It was automatically replaced: {}",
+                        template);
+            }
+
+            result.put(FILE_NAME_TEMPLATE_CONFIG, template);
+
+            return result;
+        } else {
+            return properties;
+        }
+    }
+
+    private void validate() {
+        final String connectionString = getString(AZURE_STORAGE_CONNECTION_STRING_CONFIG);
+
+        if (connectionString == null) {
+            throw new ConfigException(
+                    String.format("The configuration %s cannot be null.", AZURE_STORAGE_CONNECTION_STRING_CONFIG));
+        }
+    }
+
+    public String getConnectionString() {
+        return getString(AZURE_STORAGE_CONNECTION_STRING_CONFIG);
+    }
+
+    public String getContainerName() {
+        return getString(AZURE_STORAGE_CONTAINER_NAME_CONFIG);
+    }
+
+    @Override
+    public CompressionType getCompressionType() {
+        return CompressionType.forName(getString(FILE_COMPRESSION_TYPE_CONFIG));
+    }
+
+    @Override
+    public List<OutputField> getOutputFields() {
+        final List<OutputField> result = new ArrayList<>();
+        for (final String outputFieldTypeStr : getList(FORMAT_OUTPUT_FIELDS_CONFIG)) {
+            final OutputFieldType fieldType = OutputFieldType.forName(outputFieldTypeStr);
+            final OutputFieldEncodingType encodingType;
+            if (fieldType == OutputFieldType.VALUE) {
+                encodingType = OutputFieldEncodingType.forName(getString(FORMAT_OUTPUT_FIELDS_VALUE_ENCODING_CONFIG));
+            } else {
+                encodingType = OutputFieldEncodingType.NONE;
+            }
+            result.add(new OutputField(fieldType, encodingType));
+        }
+        return result;
```

Comment:
nit: this looks like it could be precomputed when instantiating/validating the configuration, avoiding re-parsing the OutputFieldEncodingType each time this is called.